src/app/features/course/course.component.ts

```diff
@@ -1,15 +1,51 @@
 import { Component, OnInit } from '@angular/core';
+import { FormArray, FormBuilder, FormGroup, Validators } from '@angular/forms';
 
 @Component({
   selector: 'app-course',
   templateUrl: './course.component.html',
   styleUrls: ['./course.component.scss']
 })
 export class CourseComponent implements OnInit {
+  public form!: FormGroup;
+  public isAddMode: boolean = true;
+  public submitted = false;
 
-  constructor() { }
+  constructor(private formBuilder: FormBuilder,) { }
+
+  public get title() {
+    return this.form.get('title');
+  }
+
+  public get authorName() {
+    return this.form.get('authorName');
+  }
+
+  public get description() {
+    return this.form.get('description');
+  }
+
+  public get duration() {
+    return this.form.get('duration');
+  }
+
+  get authors() : FormArray {
+    return this.form.get("authors") as FormArray
+  }
 
   ngOnInit(): void {
+    this.form = this.formBuilder.group ({
+      title: ['', Validators.required],
+      authorName: ['', Validators.pattern('^[a-zA-Z0-9]*$')],
+      description: ['', Validators.required],
+      duration: ['', Validators.required, Validators.minLength(0)],
+      authors: this.formBuilder.array([]) ,
+    })
```

Comment:
I am not sure that `ngOnInit` is a good place for form building. Lets move it into constructor or `form` field initialization.

Reply:
done